cli/src/cli.rs

```diff
@@ -537,32 +539,50 @@ impl Command {
                 }
 
                 let deployer = Signer::Local(LocalSigner::new(args.private_key));
-
                 let client = EthClient::new(&rpc_url)?;
 
-                let init_code = if !args._args.is_empty() {
-                    let init_args = parse_contract_creation(args._args)?;
-                    [args.bytecode, init_args].concat().into()
+                let bytecode = if let Some(bytecode) = args.bytecode {
+                    bytecode
+                } else {
+                    deploy_contract_from_path(args.clone()).await?
+                };
+                let init_args = if !args._args.is_empty() {
+                    parse_contract_creation(args._args)?
                 } else {
-                    args.bytecode
+                    Bytes::new()
                 };
 
-                let (tx_hash, deployed_contract_address) = deploy(
-                    &client,
-                    &deployer,
-                    init_code,
-                    Overrides {
-                        value: args.value.into(),
-                        nonce: args.nonce,
-                        chain_id: args.chain_id,
-                        gas_limit: args.gas_limit,
-                        max_fee_per_gas: args.max_fee_per_gas,
-                        max_priority_fee_per_gas: args.max_priority_fee_per_gas,
-                        ..Default::default()
-                    },
-                    true,
-                )
-                .await?;
+                let (tx_hash, deployed_contract_address) = if args.create2 {
+                    let Some(salt) = args.salt else {
+                        return Err(eyre::eyre!("Salt is required when using CREATE2"));
+                    };
```

Comment:
Instead of having a `create2` and `salt` options, which can lead to cases like this, we can take `salt = Some(...)` as create2 deploy and `salt = None` as create deploy